src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
As for the configuration, we are using the inlined version `.UseConfiguration(WebApplicationConfiguration.GetDefault(args))` in our samples. Since we want to use the settings in the "hosting.json". For the suggestion above, we would want something like `.UseConfiguration(hostingJsonFile: WebApplicationConfiguration.HostingJsonFile, commandLineArgs: args)`. 

@Tratcher did suggest adding an overload for `.UsingConfiguration(args)` which by default will call `GetDefault(args)`. This does make things more concise but at the cost of the addition of settings in hosting.json and environment variables being non-transparent.